src/Service/EventLog/Loggers/EventCache.cs

```diff
@@ -0,0 +1,11 @@
+using Passwordless.Service.EventLog.Models;
+
+namespace Passwordless.Service.EventLog.Loggers;
+
+public class EventCache
+{
+    private readonly List<ApplicationEvent> _events = new();
+    public IEnumerable<ApplicationEvent> GetEvents() => _events;
+    public bool IsEmpty() => !_events.Any();
+    public void Add(ApplicationEvent applicationEvent) => _events.Add(applicationEvent);
+}
```

Comment:
Technically it would get emptied when it gets deleted by garbage collection when the request ends (since its scoped), but it should be explicit. Making the change now.